VotingApplication/VotingApplication.Web/Scripts/Directives/VotingStrategy.js

```diff
@@ -0,0 +1,16 @@
+﻿(function () {
+    var VotingApp = angular.module('VotingApp');
+
+    VotingApp.directive('votingStrategy', ['PollStrategy', function (PollStrategy) {
+        return {
+            replace: true,
+
+            link: function (scope, element, attrs) {
+                scope.hasVotingTemplate = PollStrategy.hasVotingTemplate;
+                scope.getVotingTemplate = PollStrategy.getVotingTemplate;
+            },
+
+            template: '<div ng-include="getVotingTemplate()" ng-show="hasVotingTemplate()"></div>'
```

Comment:
I am not sure this hasVotingStrategy is really required. Surely checking if getVotingTemplate() is defined is enough?